lib/src/over_react_test/custom_matchers.dart

```diff
@@ -337,3 +339,139 @@ Matcher throwsPropError_Combination(String propName, String prop2Name, [String m
       ))
   ));
 }
+
+/// A log matcher that captures logs and uses a provided matcher or string to compare
+/// against.
+///
+/// The primary use case of the matcher is to take in a callback as the actual,
+/// and pass it to [recordConsoleLogs] to run the function and record the resulting
+/// logs that are emitted during the function runtime.
+class _LoggingFunctionMatcher extends CustomMatcher {
+  /// [matcher] is tested against the list of recorded logs
+  // could make wrapMatcher be `contains` by default as well if desired
+  _LoggingFunctionMatcher(dynamic matcher, {this.config, description, name})
+      : super(description ?? 'emits the logs', name ?? 'logs', _customWrapMatcher(matcher));
+
+  final ConsoleConfiguration config;
+
+  static Matcher _customWrapMatcher(dynamic x) {
+    if (x == null || (x is List && x.isEmpty)) {
+      throw ArgumentError('The expected value cannot be null or an empty list. \n\n'
+          'If you are expecting no logs, use the emitsNoLogs matcher');
+    }
+
+    if (x is Matcher) {
+      return x;
+    } else if (x is String) {
+      return anyElement(contains(x));
+    } else {
+      return equals(x);
+    }
+  }
+
+  @override
+  featureValueOf(actual) {
+    var logs = <String>[];
+
+    if (actual is List) return actual;
+
+    if (actual is! Function()) {
+      throw ArgumentError('The actual value must be a callback or a List.');
+    }
+
+    try {
+      logs = recordConsoleLogs(actual, config ?? logConfig);
+    } catch (_) {}
+
+    return logs;
+  }
+}
+
+/// A Matcher used to compare a list of logs against a provided matcher.
+///
+/// __Examples:__
+///
+///     When passed a string, the matcher look for any log that contains that
```

Comment:
These need to be unindented since right now they're interpreted as markdown code blocks
<img width="631" alt="Screen Shot 2019-11-15 at 1 39 19 PM" src="https://user-images.githubusercontent.com/1713967/68974195-5c730580-07ad-11ea-97de-9b9b77425337.png">